sdk/cargo-build-bpf/src/main.rs

```diff
@@ -56,18 +57,125 @@ where
         .args(&args)
+        .stdout(Stdio::piped())
         .spawn()
         .unwrap_or_else(|err| {
             eprintln!("Failed to execute {}: {}", program.display(), err);
             exit(1);
         });
 
-    let exit_status = child.wait().expect("failed to wait on child");
-    if !exit_status.success() {
+    let output = child.wait_with_output().expect("failed to wait on child");
+    if !output.status.success() {
         exit(1);
     }
+    output
+        .stdout
+        .as_slice()
+        .iter()
+        .map(|&c| c as char)
+        .collect::<String>()
+}
+
+// Check whether a package is installed and install it if it is missing.
+fn install_if_missing(
+    config: &Config,
+    package: &str,
+    version: &str,
+    url: &str,
+    file: &Path,
+) -> Result<(), String> {
+    // Check whether the package is already in ~/.cache/solana.
+    // Donwload it and place in the proper location if not found.
+    let home_dir = PathBuf::from(env::var("HOME").unwrap_or_else(|err| {
+        eprintln!("Can't get home directory path: {}", err);
+        exit(1);
+    }));
+    let target_path = home_dir
+        .join(".cache")
+        .join("solana")
+        .join(version)
+        .join(package);
+    if !target_path.is_dir() {
+        if target_path.exists() {
+            fs::remove_file(&target_path).map_err(|err| err.to_string())?;
+        }
+        let mut url = String::from(url);
+        url.push('/');
+        url.push_str(version);
+        url.push('/');
+        url.push_str(file.to_str().unwrap());
+        download_file(&url.as_str(), &file, false)?;
+        fs::create_dir_all(&target_path).map_err(|err| err.to_string())?;
+        let tar = PathBuf::from("tar");
+        let tar_args = vec![
+            "-xjf",
+            file.to_str().unwrap(),
+            "-C",
+            target_path.to_str().unwrap(),
+            "--strip-components",
+            "1",
+        ];
+        spawn(&tar, &tar_args);
+        fs::remove_file(file).map_err(|err| err.to_string())?;
+    }
+    // Make a symbolyc link source_path -> target_path in the
+    // sdk/bpf/dependencies directory if no valid link found.
+    let source_base = config.bpf_sdk.join("dependencies");
+    if !source_base.exists() {
+        fs::create_dir_all(&source_base).map_err(|err| err.to_string())?;
+    }
+    let source_path = source_base.join(package);
+    // Check whether the correct symbolic link exists.
+    let missing_source = if source_path.exists() {
+        let invalid_link = if let Ok(link_target) = source_path.read_link() {
+            link_target != target_path
+        } else {
+            true
+        };
+        if invalid_link {
+            fs::remove_file(&source_path).map_err(|err| err.to_string())?;
+        }
+        invalid_link
+    } else {
+        true
+    };
+    if missing_source {
+        std::os::unix::fs::symlink(target_path, source_path).map_err(|err| err.to_string())?;
```

Comment:
We build `cargo-build-bpf` for windows too :)

Reply:
fixed